runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine.kt

```diff
@@ -26,14 +26,17 @@ import okhttp3.TlsVersion as OkHttpTlsVersion
  * [aws.smithy.kotlin.runtime.http.engine.HttpClientEngine] based on OkHttp.
  */
 public class OkHttpEngine(
-    private val config: OkHttpEngineConfig,
+    override val config: OkHttpEngineConfig,
 ) : HttpClientEngineBase("OkHttp") {
     public constructor() : this(OkHttpEngineConfig.Default)
 
     public companion object {
```

Comment:
**suggestion**: This `companion` (and `CrtHttpEngine.Companion`) could just implement `EngineFactory` interface removing the need for a separate `OkHttp`/`Crt` objects.

e.g.

```kotlin
public class OkHttpEngine(...) : HttpClientEngineBase("OkHttp") {
     ...

     public companion object : EngineFactory<OkHttpEngineConfig.Builder, OkHttpEngine> {
         override val engineConstructor: (OkHttpEngineConfig.Builder.() -> Unit) -> OkHttpEngine = ::invoke
         
            public operator fun invoke(block: OkHttpEngineConfig.Builder.() -> Unit): OkHttpEngine = ...
    }
}

```

This allows the engine itself to be passed to the `httpClient` config methods:

```kotlin
    StsClient {
        httpClient(OkHttpEngine) {
            ...
        }
    }
```

This works because it will infer the `companion` object automatically.

Reply:
Will do.